### ISSUE TITLE
Need to upgrade xmlhttprequest-ssl for security issue

### ISSUE THREAD

**Issue by henriquef-cit** [artilleryio/artillery#1018]
Describe the bug
`npm audit` is failing due to a vulnerability in `xmlhttprequest-ssl` which appears to have been patched already. Its a dependency of engine.io-client.
[see npm advisory](https://www.npmjs.com/advisories/1665)

┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Arbitrary Code Injection                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ xmlhttprequest-ssl                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=1.6.2                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ artillery [dev]                                              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ artillery > socket.io-client > engine.io-client >            │
│               │ xmlhttprequest-ssl                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1665                            │
└───────────────┴──────────────────────────────────────────────────────────────┘

**Comment by hassy** [comment]
Thank you @henriquef-cit, on it! 🛡️

**Comment by idealley** [comment]
Additional security errors were raised:

```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ high          │ Arbitrary Code Injection                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ xmlhttprequest-ssl                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=1.6.2                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ artillery                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ artillery > socket.io-client > engine.io-client >            │
│               │ xmlhttprequest-ssl                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1665                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ high          │ Command Injection                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ lodash                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.17.21                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ artillery                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ artillery > artillery-plugin-statsd > lodash                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1673                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ high          │ Command Injection                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ lodash                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.17.21                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ artillery                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ artillery > cheerio > lodash                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1673                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ high          │ Command Injection                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ lodash                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.17.21                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ artillery                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ artillery > lodash                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1673                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ high          │ Arbitrary Code Execution                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ underscore                                                   │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=1.12.1                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ artillery                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ artillery > jsonpath > underscore                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1674                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
5 vulnerabilities found - Packages audited: 270
```

**Comment by hlhl040** [comment]
Hi, when will we release a new version that have dependency of jsonpath ^1.1.1?
